reactome_analysis_api/reactome_analysis_api/searcher/public_data_searcher.py

```diff
@@ -0,0 +1,125 @@
+        """
+        The init is used to define the path, the path is set as an enviroment variable
+        """
+        self._path = path
+
+    def setup_search_events(self):  # top level funktion hinzufpügen create_search_index
+        """
+        sets up the index for later search process based on schema, sets up species for later filtering
+        """
+        LOGGER.info("Creating index for searching")
+        if not os.path.exists(path=self._path):
+            os.mkdir(self._path)
+
+        ix = create_in(self._path, self.schema)
+        LOGGER.info("Created index: ", self._path)
+        writer = ix.writer()
+        LOGGER.info("Fetching available datasets from GREIN")
+        grein_datasets = Fetcher.get_available_datasets_grein(10)
+        for dataset in grein_datasets:
+            writer.add_document(data_source="grein", id=str(dataset['id']), title=str(dataset['title']),
+                                species=str(dataset['species']),
+                                description=str(dataset['study_summary']), no_samples=str(dataset['no_samples']),
+                                technology=str(dataset['technology']),
+                                resource_id=str(dataset['resource_id']),
+                                loading_parameters=str(dataset['loading_parameters']))
+
+        LOGGER.info("Fetching available datasets from ExpressionAtlas")
+        expression_atlas_datasets = Fetcher.get_available_datasets_expression_atlas()
+        for dataset in expression_atlas_datasets:
+            writer.add_document(data_source="ebi_gxa", id=str(dataset['id']), title=str(dataset['title']),
+                                species=str(dataset['species']),
+                                description=str(''), no_samples=dataset['no_samples'],
+                                technology=str(dataset['technology']),
+                                resource_id=str(dataset['resource_id']),
+                                loading_parameters=str(dataset['loading_parameters']))
+        writer.commit()
+        list_data = grein_datasets + expression_atlas_datasets
+        species_in_datasets = self._get_species(list_data)  # gets species based on public datasets
+        with open(self._path + 'species.pickle', 'wb') as f:
+            pickle.dump(species_in_datasets, f, pickle.HIGHEST_PROTOCOL)
+
+    def _get_species(self, datasets) -> set:
+        """
+        :param datasets: list of dictionaries from public datasets
+        :return species_values: list of species in public datasets
+        """
+        values = set()
+        for dictionary in datasets:
+            if 'species' in dictionary:
+                values.add(dictionary['species'])
+        species_values = sorted(values)
+        return species_values
+
+    def get_species(self) -> list:
+        """
+        returns species stored in binary file
+        """
+        try:
+            if os.path.exists(self._path + 'species.pickle'):
```

Comment:
Here, I'd add some kind of cache. Then, the data only has to be loaded the first time and we'll save some additional time.